react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt

```diff
@@ -96,141 +92,155 @@ class SyrfClientModule(reactContext: ReactApplicationContext) :
     permissions: Array<out String>,
     grantResults: IntArray
   ): Boolean {
-    PermissionsManager(activity).handleResults(
-      permissions,
-      successCallback = {
-        if (waitingForLocationPermission) {
-          SYRFLocation.subscribeToLocationUpdates(activity)
-        }
-        waitingForLocationPermission = false
-        if (waitingForCurrentLocationPermission) {
-          getCurrentLocation()
-        }
-        waitingForCurrentLocationPermission = false
-      },
-      exceptionCallback = {
-        waitingForLocationPermission = false
-      })
-
-    return true
+    if (currentActivity != null) {
+      PermissionsManager(currentActivity!!).handleResults(
+        permissions,
+        successCallback = {
+          if (waitingForLocationPermission) {
+            SYRFLocation.subscribeToLocationUpdates(currentActivity!!)
+          }
+          waitingForLocationPermission = false
+          if (waitingForCurrentLocationPermission) {
+            getCurrentLocation()
+          }
+          waitingForCurrentLocationPermission = false
+        },
+        exceptionCallback = {
+          waitingForLocationPermission = false
+        })
+      return true
+    }
+    return false
   }
 
   @ReactMethod
   fun configure(params: ReadableMap, promise: Promise) {
-
-    val builder = SYRFLocationConfig.Builder()
-    getLongOrNull(params, KEY_UPDATE_INTERVAL)?.let {
-      builder.updateInterval(it)
-    }
-    getIntOrNull(params, KEY_MAX_LOCATION_ACCURACY)?.let {
-      builder.maximumLocationAccuracy(it)
-    }
-    getMapOrNull(params, KEY_PERMISSION_REQUEST_CONFIG)?.let { permissionRequestParams ->
-      val permissionConfigBuilder = SYRFPermissionRequestConfig.Builder()
-
-      getStringOrNull(permissionRequestParams, KEY_PERMISSION_REQUEST_TITLE)?.let {
-        permissionConfigBuilder.title(it)
-      }
-      getStringOrNull(permissionRequestParams, KEY_PERMISSION_REQUEST_MESSAGE)?.let {
-        permissionConfigBuilder.message(it)
+    if (currentActivity != null) {
+      val builder = SYRFLocationConfig.Builder()
+      getLongOrNull(params, KEY_UPDATE_INTERVAL)?.let {
+        builder.updateInterval(it)
       }
-      getStringOrNull(permissionRequestParams, KEY_PERMISSION_REQUEST_OK_BTN)?.let {
-        permissionConfigBuilder.okButton(it)
+      getIntOrNull(params, KEY_MAX_LOCATION_ACCURACY)?.let {
+        builder.maximumLocationAccuracy(it)
       }
-      getStringOrNull(permissionRequestParams, KEY_PERMISSION_REQUEST_CANCEL_BTN)?.let {
-        permissionConfigBuilder.cancelButton(it)
+      getMapOrNull(params, KEY_PERMISSION_REQUEST_CONFIG)?.let { permissionRequestParams ->
```

Comment:
here we can extract permissionConfigBuilder into a separate function with all configs.